<?xml version="1.0"?>
<launch>
    <arg name="camera_id" default="alexa"/>
    <arg name="topic_adapter" default="/realm/$(arg camera_id)/input"/>
    <arg name="topic_pose_est" default="/realm/$(arg camera_id)/pose_estimation/frame"/>
    <arg name="topic_dense" default="/realm/$(arg camera_id)/densification/frame"/>
    <arg name="topic_surf" default="/realm/$(arg camera_id)/surface_generation/frame"/>
    <arg name="topic_rect" default="/realm/$(arg camera_id)/ortho_rectification/frame"/>
    <arg name="topic_mosaic" default="/realm/$(arg camera_id)/mosaicing/frame"/>

    <node pkg="realm_ros" type="realm_exiv2_grabber" name="realm_exiv2_grabber" output="screen">
        <param name="config/id" type="string" value="$(arg camera_id)"/>
        <param name="config/input" type="string" value="/media/hdd/Datasets/edm_big_overlap_50p"/>
        <param name="config/rate" type="double" value="10.0"/>
        <param name="config/profile" type="string" value="alexa_noreco"/>
        <!--param name="config/opt/poses" type="string" value=""/-->
        <!--param name="config/opt/surface_pts" type="string" value=""/-->
        <!--param name="config/opt/set_all_keyframes" type="bool" value="True"/-->
        <!--param name="config/opt/working_directory" type="string" value="/home/alex/Documents/OpenREALM"/-->
    </node>

    <node pkg="realm_ros" type="realm_stage_node" name="realm_pose_estimation" output="screen">
        <param name="stage/type" type="string" value="pose_estimation"/>
	<param name="stage/master" type="bool" value="true"/>
        <param name="topics/input/frame" type="string" value="$(arg topic_adapter)"/>
        <param name="topics/output" type="string" value="$(arg topic_pose_est)"/>
        <param name="config/id" type="string" value="$(arg camera_id)"/>
        <param name="config/profile" type="string" value="alexa_noreco"/>
        <param name="config/method" type="string" value="ov2slam"/>
        <!--param name="config/opt/working_directory" type="string" value="/home/alex/Documents/OpenREALM"/-->
        <!--param name="config/opt/output_directory" type="string" value="/home/alex/Documents/output_test"/-->
    </node>

    <node pkg="realm_ros" type="realm_stage_node" name="realm_surface_generation" output="screen">
        <param name="stage/type" type="string" value="surface_generation"/>
        <param name="topics/input/frame" type="string" value="$(arg topic_pose_est)"/>
        <param name="topics/output" type="string" value="$(arg topic_surf)"/>
        <param name="config/id" type="string" value="$(arg camera_id)"/>
        <param name="config/profile" type="string" value="alexa_noreco"/>
        <!--param name="config/opt/working_directory" type="string" value="/home/alex/Documents/OpenREALM"/-->
        <!--param name="config/opt/output_directory" type="string" value="/home/alex/Documents/output_test"/-->
    </node>

    <node pkg="realm_ros" type="realm_stage_node" name="realm_ortho_rectification" output="screen">
        <param name="stage/type" type="string" value="ortho_rectification"/>
        <param name="topics/input/frame" type="string" value="$(arg topic_surf)"/>
        <param name="topics/output" type="string" value="$(arg topic_rect)"/>
        <param name="config/id" type="string" value="$(arg camera_id)"/>
        <param name="config/profile" type="string" value="alexa_noreco"/>
        <!--param name="config/opt/working_directory" type="string" value="/home/alex/Documents/OpenREALM"/-->
        <!--param name="config/opt/output_directory" type="string" value="/home/alex/Documents/output_test"/-->
    </node>

    <node pkg="realm_ros" type="realm_stage_node" name="realm_mosaicing" output="screen">
        <param name="stage/type" type="string" value="mosaicing"/>
        <param name="topics/input/frame" type="string" value="$(arg topic_rect)"/>
        <param name="config/id" type="string" value="$(arg camera_id)"/>
        <param name="config/profile" type="string" value="alexa_noreco"/>
        <!--param name="config/opt/working_directory" type="string" value="/home/alex/Documents/OpenREALM"/-->
        <!--param name="config/opt/output_directory" type="string" value="/home/alex/Documents/output"/-->
    </node>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find realm_ros)/rviz/realm.rviz" />



</launch>
